File: data/code_pairs/cpp/34.cpp
#include <iostream>
#include <cmath> // For fabs function

class eapVarApp {
public:
    static float f(float x) {
        float j, k, l;
        j = x;
        k = g(j * j);
        l = j * g(j * j * j);
        return k;
    }
    
    static float g(float x) {
        return x;
    }
    
    static float h(float x) {
        return f(x);
    }
};

File: data/code_pairs/fortran/34.f90
module eapVarApp
contains
    function f(x)
        real :: x
        real :: f
        real :: j, k, l
        j = x
        k = g(j*j)
        l = j * g(j * j * j)
        f = k
    end function f

    function g(x)
        real :: x
        real :: g
        g = x
    end function g

    function h(x)
        real :: x
        real :: h
        h = f(x)
    end function h
end module eapVarApp